Add comprehensive evaluation framework collateral

gsong opened the pull request:
## Summary

Add practical collateral to help teams select and implement AI evaluation approaches:

- **Quick Assessment Tool**: 2-minute questionnaire providing instant evaluation recommendations
- **Cost-Benefit Calculator**: ROI worksheets with industry examples and business case templates
- **Implementation Roadmaps**: Ready-to-use project plans for 4 scenarios (startup, enterprise, research, emergency)
- **Red Flag Checklist**: Early warning system for evaluation gaps with urgency-based categorization
- **Success Stories**: 5 detailed case studies with concrete ROI data across industries
- **Tool Comparison Matrix**: Comprehensive comparison of 30+ evaluation tools with decision guidance

## Test plan

- [ ] Review each new file for completeness and accuracy
- [ ] Validate that tools provide actionable guidance for different team scenarios
- [ ] Ensure consistent formatting and cross-references between documents
- [ ] Test quick assessment tool recommendations against known use cases
- [ ] Verify cost calculations and ROI examples are realistic

gsong committed:
feat: add comprehensive evaluation framework collateral

- Add quick assessment tool for instant evaluation approach recommendations
- Add cost-benefit calculator with ROI worksheets and business case templates
- Add implementation roadmap templates for different project scenarios
- Add red flag checklist for early warning of evaluation gaps
- Add success stories with concrete ROI data across 5 industries
- Add comprehensive tool comparison matrix with decision guidance

These practical tools bridge the gap between evaluation theory and
implementation, helping teams quickly select and justify the right
evaluation approaches for their AI projects.

oshinanand commented on decision-trees/metric-selection-by-task.md:
```diff
@@ -0,0 +1,280 @@
+
+- **Answer Accuracy** (Weight: 25%)
+  - Measurement: LLM-as-judge with verification prompts
+  - Target: >90% accuracy for factual questions
```
Should be all other metrics ( depending on what accuracy is )
Metrics 
referencing (knowledge grounding) 
benchmarking
and human eval

oshinanand replied:
Where are we getting this number from : generative task , putting a number like this might be tricky;

oshinanand commented on decision-trees/metric-selection-by-task.md:
```diff
@@ -0,0 +1,280 @@
+  - Target: >90% accuracy for factual questions
+- **Relevance to Query** (Weight: 15%)
+  - Measurement: Semantic similarity + human validation
+  - Target: >85% relevance score
```
same as above in the target reference

oshinanand commented on decision-trees/metric-selection-by-task.md:
```diff
@@ -0,0 +1,280 @@
+  - Measurement: End-to-end latency tracking
+  - Target: <2 seconds p95
+- **Clarity & Readability** (Weight: 10%)
+  - Measurement: Flesch-Kincaid + human assessment
```
Just a note:-These readability scores haven't worked great in the past with generative task

oshinanand commented on decision-trees/metric-selection-by-task.md:
```diff
@@ -0,0 +1,280 @@
+**Priority 1 (Content Quality - 50% of evaluation budget)**
+
+- **Faithfulness to Source** (Weight: 25%)
+  - Measurement: Factual consistency checking
```
previous summary cases : there might additionally be target user definition that defines success criteria

gsong committed:
feat: add LLM evaluation metrics selection guide

Transform evaluation from comprehensive taxonomy to practical selection menu:
- Quick start metric bundles by use case (RAG, customer service, creative content)
- Quality dimension framework with coverage checklist
- Implementation difficulty indicators and resource-based guidance
- Phased rollout strategy and anti-patterns to avoid
- Tool recommendations and practical implementation checklists

Enables teams to select focused metric sets based on application needs rather than measuring everything.

gsong committed:
refactor: restructure evaluation guides to eliminate overlap

- Transform metrics-selection-guide.md into focused reference catalog
- Update evaluation-selection-wizard.md to emphasize interactive guidance
- Add cross-references between complementary documents
- Remove redundant implementation timelines and budget details

gsong committed:
docs: add LM generative task evaluation mindnode

- Include visual mindmap for evaluation framework structure
- Provides alternative navigation and conceptual overview

oshinanand commented on automation-templates/pipeline-templates.md:
```diff
@@ -0,0 +1,1117 @@
+  name: "basic_text_evaluation"
+  version: "1.0.0"
+
+input:
```
Why do we need to bring in LLM eval here

oshinanand commented on automation-templates/pipeline-templates.md:
````diff
@@ -0,0 +1,1117 @@
+        return json.loads(completion.choices[0].message.content)
+```
+
+### 1.3 Production Monitoring Pipeline
````
Discussion on this
